jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmapper/FailureExceptionMapper.java

```diff
@@ -55,14 +55,28 @@ public Response toResponse(Failure exception) {
       logger.error("An unhandled exception was thrown.", exception);
     }
 
-    return Response
-        .status(status)
-        .entity(
-            JsonError.builder()
-                .code(status.getStatusCode())
-                .message(status.getReasonPhrase())
-                .build())
-        .type(MediaType.APPLICATION_JSON)
-        .build();
+    Response response = exception.getResponse();
```

Comment:
Rather than having mostly the same logic in each of the conditional cases maybe this could be simplified with a private method.

Something like below (note I'm not totally set on this and would like other opinions before this change is made).
```java
return responseWithStatus(exception, status)
     .entity(
         JsonError.builder()
             .code(code)
             .message(status.getReasonPhrase())
             .build())
     .type(MediaType.APPLICATION_JSON)
     .build();


private Response responseWithStatus(Failure e, Response.Status status) {
  return exception.getResponse() != null ? Response.fromResponse(exception.getResponse()) : Response.status(status);
}
```

Reply:
I agree that this can definitely be condensed a bit. This is also the case for WebApplicationExceptionMapper.

Reply:
I can't really think of a better method. I do find it strange though that the method is called responseWithStatus and the user passes in a Status, but the Status isn't used if the Failure has a response object.

Reply:
What about something like:

```java
return Optional.ofNullable(exception.getResponse())
    .map(Response::fromResponse)
    .orElse(Response.status(status))
    .entity(
        JsonError.builder()
            .code(code)
            .message(status.getReasonPhrase())
            .build())
    .type(MediaType.APPLICATION_JSON)
    .build();
```

Reply:
+1 to @b-boogaard suggestion.

Reply:
Settled on this: https://github.com/cerner/beadledom/pull/45/commits/cf92faf55674bfa402abe51eff5459464d3ba663#diff-a3d460ba2485f948686ff5823fe46af8R49